torchgeo/datasets/geo.py

```diff
@@ -896,21 +903,6 @@ def __getitem__(self, query: GeoSlice) -> dict[str, Any]:
 
         return sample
 
-    def get_label(self, feature: 'fiona.model.Feature') -> int:
```

Comment:
Not sure if we need to deprecate this or not. This is not as performant as using geopandas directly as this function requires a for loop and my change using geopandas doesn't. The only gotcha here is that the Eurocrops dataset overrides this method with some custom logic.

Reply:
It is [currently documented](https://torchgeo.readthedocs.io/en/stable/api/datasets.html#torchgeo.datasets.VectorDataset.get_label), so it's technically a public API function. If we can find a way to deprecate it, that would be ideal. If we can't, well that's what the major version bump is for. I'll mark this as currently backwards-incompatible for now.

Reply:
I elected to keep it in as I can see why it's useful if a dataset needs some custom logic to convert the labels to class indices